### PR TITLE
Release candidate 4.5.01

### PR DESCRIPTION
**Pull request by crtrott**
* Fix build issue with ninja and use of make clean [\#7557](https://github.com/kokkos/kokkos/pull/7557)
* Update mdspan to include fix for submdspan and bracket operator with clang 15&16 [\#7559](https://github.com/kokkos/kokkos/pull/7559)
* Fix build issues and a test with latest MSVC/CUDA compiler versions [\#7566](https://github.com/kokkos/kokkos/pull/7566)
* Fix DynRankView performance regression by re-introducing shortcut operator() impls [\#7606](https://github.com/kokkos/kokkos/pull/7606)

Also updates version numbers, separate single commit for each of the things.

### CODE REVIEW

**Review by masterleinad — APPROVED** [review]

File: core/unit_test/TestViewBadAlloc.hpp
[the remark this answers is not in the document]
Comment (crtrott) [reply core/unit_test/TestViewBadAlloc.hpp:1]: Yeah the problem was that the tests wouldn't build with the newer MSVC, so it's hard to verify the Kokkos version works.

**Review by ndellingwood — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,16 @@
...
 
+## 4.5.01
+
+[Full Changelog](https://github.com/kokkos/kokkos/compare/4.5.00...4.5.01)
+
+### Bug Fixes
+
+* Fix build issue with ninja and use of make clean [\#7557](https://github.com/kokkos/kokkos/pull/7557)
+* Update mdspan to include fix for submdspan and bracket operator with clang 15&16 [\#7559](https://github.com/kokkos/kokkos/pull/7559)
+* Fix build issues and a test with latest MSVC/CUDA compiler versions [\#7566](https://github.com/kokkos/kokkos/pull/7566)
```
Comment (dalg24) [inline CHANGELOG.md:11]:
```suggestion
```
No one cares

**Review by ndellingwood — APPROVED** [review]

**Review by ndellingwood — APPROVED** [review]

**Comment by dalg24** [comment]
You need to cherry-pick #7573 as well for the CI to run

**Comment by ndellingwood** [comment]
CI job failure summary:

**linux-x64 / CI (ubuntu:latest, g++, RelWithDebInfo, THREADS...)** - issues downloading google benchmark
```
[benchmark] CMake Error at googlebenchmark-subbuild/googlebenchmark-populate-prefix/src/googlebenchmark-populate-stamp/download-googlebenchmark-populate.cmake:170 (message):
[benchmark]   Each download failed!
[benchmark] 
[benchmark]     error: downloading 'https://github.com/google/benchmark/archive/refs/tags/v1.7.1.tar.gz' failed
[benchmark]           status_code: 22
[benchmark]           status_string: "HTTP response code said error"
[benchmark]           log:
[benchmark]           --- LOG BEGIN ---
[benchmark]           Host github.com:443 was resolved.
```

**ci/gitlab-ci INTEL-DATA-CENTER-MAX-1100** - various occurrences of `Native API failed` in tests
```
40: [ RUN      ] sycl.scatterview
40: terminate called after throwing an instance of 'std::runtime_error'
40:   what():  There was a synchronous SYCL error:
40: Native API failed. Native API returns: -2 (PI_ERROR_DEVICE_NOT_AVAILABLE) -2 (PI_ERROR_DEVICE_NOT_AVAILABLE)
...
45: [ RUN      ] std_algorithms_non_mod_seq_ops.search
45: unknown file: Failure
45: C++ exception with description "There was a synchronous SYCL error:
45: Native API failed. Native API returns: -2 (PI_ERROR_DEVICE_NOT_AVAILABLE) -2 (PI_ERROR_DEVICE_NOT_AVAILABLE)" thrown in the test body.
45: [  FAILED  ] std_algorithms_non_mod_seq_ops.search (8124 ms)
45: [ RUN      ] std_algorithms_non_mod_seq_ops.search_n
45: unknown file: Failure
45: C++ exception with description "Native API failed. Native API returns: -5 (PI_ERROR_OUT_OF_RESOURCES) -5 (PI_ERROR_OUT_OF_RESOURCES)" thrown in the test body.
45: [  FAILED  ] std_algorithms_non_mod_seq_ops.search_n (1001 ms)
45: [----------] 4 tests from std_algorithms_non_mod_seq_ops (12311 ms total)
45: 
45: [----------] 1 test from std_algorithms_count_test
45: [ RUN      ] std_algorithms_count_test.test
45: Enqueue process failed. -59 (PI_ERROR_INVALID_OPERATION)
45: terminate called after throwing an instance of 'std::runtime_error'
45:   what():  There was an asynchronous SYCL error!
...
```

**jenkins/pr-merge HIP-ROCm-5.2** - Docker image issues
```
#6 4.384 E: The repository 'https://repo.radeon.com/amdgpu/22.20/ubuntu focal Release' does not have a Release file.

#6 ERROR: process "/bin/sh -c apt-get update && apt-get install -y         kmod         wget         ccache         file         &&     apt-get clean &&     rm -rf /var/lib/apt/lists/*" did not complete successfully: exit code: 100
...
```

**Comment by dalg24** [comment]
These are all unrelated

**Pull request merged**